example/pages/index.js

```diff
@@ -15,4 +15,13 @@ const Home = () => (
   </div>
 );
 
-export default withUrqlClient({ url: 'https://graphql-pokemon.now.sh' })(Home);
+export default withUrqlClient(ctx => {
+  return {
+    url: 'https://graphql-pokemon.now.sh',
+    fetchOptions: {
+      headers: {
+        Authorization: `Bearer ${ctx.req.token}`,
+      },
+    },
+  };
+})(Home);
```

Comment:
Our API working correctly! The `ctx` accessor function _only_ gets run inside of `getInitialProps`, so we don't run into any circular structure issues. In our example we're not actually passing any sort of `token`, this is more just validating that a user _could_ safely access `ctx.req`.